src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -174,12 +174,12 @@ private void SetGlobalErrorIfTrue(bool arg)
             if (filterOpt == null)
             {
                 WarnUnusedFields(compilation, diagnostics, cancellationToken);
-            }
 
-            MethodSymbol entryPoint = GetEntryPoint(compilation, moduleBeingBuiltOpt, hasDeclarationErrors, diagnostics, cancellationToken);
-            if (moduleBeingBuiltOpt != null && entryPoint != null && compilation.Options.OutputKind.IsApplication())
-            {
-                moduleBeingBuiltOpt.SetPEEntryPoint(entryPoint, diagnostics);
+                MethodSymbol entryPoint = GetEntryPoint(compilation, moduleBeingBuiltOpt, hasDeclarationErrors, diagnostics, cancellationToken);
```

Comment:
The implication of this change is that if `filterOpt` is specified we will no longer get diagnostics about Main, right?  What is that set?  Just "no main", and "ambiguous main"?